### PR TITLE
Decompose archive smoke-tests

### PR DESCRIPTION
**Pull request by dlvenable**
### Description

The smoke tests for archive files (.tar.gz) are not always consistent and then run in sequence. This makes it difficult to re-run any given job.

This PR decomposes the smoke-tests such that each execution runs against a specific base image and tests a specific archive distribution (e.g. JDK vs. no JDK). Additionally, it updates the release action to run each different target as its own job within the action. This way, any given job can easily be re-run. It also improves the visibility into these tests.

You can see a sample run from my fork here: https://github.com/dlvenable/data-prepper/actions/runs/3132049489
 
### Issues Resolved
N/A
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: release/smoke-tests/run-tarball-files-smoke-tests.sh [hunk 1]
```diff
@@ -146,12 +142,14 @@ function run_smoke_test() {
     cd "${CURRENT_DIR}" || exit
 }
 
-while getopts "b:d:hn:u:v:" arg; do
+while getopts "b:d:hi:n:t:u:v:" arg; do
```
Comment (sbayer55) [inline release/smoke-tests/run-tarball-files-smoke-tests.sh:145]: Should `-i <string>` and `-t <string>` be a required argument? On lines 194-202 it looks like FROM_IMAGE and TAR_NAME have default values.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -72,7 +72,17 @@ jobs:
           docker pull ${{ secrets.ECR_REPOSITORY_URL }}:${{ env.version }}-${{ github.run_number }}
           ./release/smoke-tests/run-smoke-tests.sh -v ${{ env.version }}-${{ github.run_number }} -i ${{ secrets.ECR_REPOSITORY_URL }}
 
-  validate-tarball:
+  validate-archive:
+    strategy:
+      matrix:
+        include:
+          - image : "openjdk:11"
```
Comment (sbayer55) [inline .github/workflows/release.yml:79]: Nice cleanup here

**Review by sbayer55 — CHANGES_REQUESTED** [review]

File: release/smoke-tests/run-tarball-files-smoke-tests.sh (hunk 1 above)
Comment (dlvenable) [reply release/smoke-tests/run-tarball-files-smoke-tests.sh:145]: No, I don't think they need to be required. Is this requiring them?

Comment (dlvenable) [reply release/smoke-tests/run-tarball-files-smoke-tests.sh:145]: I have been able to run this script without `-i` and `-t`.

Comment (sbayer55) [reply release/smoke-tests/run-tarball-files-smoke-tests.sh:145]: Thanks for confirming `-i` and `-t` are actually optional. Checking the `getops` documentation confirms an option character followed by a `:` is expecting an argument if the flag is present.

If curious, the official documentation is available here:
https://man7.org/linux/man-pages/man1/getopts.1p.html

**Review by sbayer55 — APPROVED** [review]

**Review by oeyh — APPROVED** [review]

**Review by sbayer55 — APPROVED** [review]

**Pull request merged**